File: solutions/golang/917/solution.go
func reverseOnlyLetters(s string) string {
	r := []rune(s)
	left := 0
	right := len(r) - 1
	for left < right {
		for !(r[left] >= 'A' && r[left] <= 'Z' ||
			r[left] >= 'a' && r[left] <= 'z') {
			left++
			if left >= right {
				break
			}
		}
		for !(r[right] >= 'A' && r[right] <= 'Z' ||
			r[right] >= 'a' && r[right] <= 'z') {
			right--
			if left >= right {
				break
			}
		}
		if left < right {
			r[left], r[right] = r[right], r[left]
			left++
			right--
		}
	}

	return string(r)
}

File: solutions/cpp/917/solution.cpp
class Solution {
public:
    string reverseOnlyLetters(string s) {
        int left = 0, right = s.size() - 1;
        while (left < right) {
            while (!(s[left] >= 'A' && s[left] <= 'Z' ||
                s[left] >= 'a' && s[left] <= 'z')) {
                left++;
                if (left >= right) {
                    break;
                }
            }
            while (!(s[right] >= 'A' && s[right] <= 'Z' ||
                s[right] >= 'a' && s[right] <= 'z')) {
                right--;
                if (left >= right) {
                    break;
                }
            }
            if (left < right) {
                swap(s[left], s[right]);
                left++;
                right--;
            }
        }
        return s;
    }
};